hass_nabucasa/ice_servers.py

```diff
@@ -0,0 +1,143 @@
+"""Manage ICE servers."""
+
+from __future__ import annotations
+
+import asyncio
+from collections.abc import Awaitable, Callable
+import logging
+import random
+import time
+from typing import TYPE_CHECKING
+
+from aiohttp import ClientResponseError
+from aiohttp.hdrs import AUTHORIZATION, USER_AGENT
+from webrtc_models import RTCIceServer
+
+if TYPE_CHECKING:
+    from . import Cloud, _ClientT
+
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class IceServers:
+    """Class to manage ICE servers."""
+
+    def __init__(self, cloud: Cloud[_ClientT]) -> None:
+        """Initialize ICE Servers."""
+        self.cloud = cloud
+        self._refresh_task: asyncio.Task | None = None
+        self._ice_servers: list[RTCIceServer] = []
+        self._ice_servers_listener: Callable[[], Awaitable[None]] | None = None
+        self._ice_servers_listener_unregister: Callable[[], None] | None = None
+
+    async def _async_fetch_ice_servers(self) -> None:
+        """Fetch ICE servers."""
```

Comment:
A function named like this, I would expect to return a list of ice servers.
I think we should:
a) Rename it to `_async_refresh_ice_servers`/ `_async_reload_ice_servers`
b) Change the logic to return the servers, and then set the response where its currently called.
_Personally I would prefer b_

Reply:
Changed the flow to b option 👌🏻